lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+    format: function (val) {
+      if (!/(debug|verbose|info|warn|error)/.test(val)) {
+        throw new Error('Loglevel has to be set to `debug`, `verbose`, `info`, `warn`, or `error`')
+      }
+    },
+    env: 'CMD_LOGLEVEL',
+    default: 'info'
+  },
+  urlAddPort: {
+    doc: 'set to add port on callback URL (ports `80` or `443` won\'t be applied) (only applied when domain is set)',
+    format: 'Boolean',
+    env: 'CMD_URL_ADDPORT',
+    default: false
+  },
+  allowOrigin: {
+    doc: 'CORS whitelist for non-API calls',
+    format: function (val) {
+      for(let i = val.length; i--;) {
+        if (!domainRegex.test(val[i])) {
+          throw new Error(`CORS array contains invalid domain name '${val[i]}'`)
+        }
+      }
+    },
+    env: 'CMD_ALLOW_ORIGIN',
+    default: ['localhost']
+  },
+  useSSL: {
+    doc: 'Set to use SSL server (if `true`, will auto turn on `protocolUseSSL`',
+    format: 'Boolean',
+    env: 'CMD_USESSL',
+    default: false
+  },
+  hsts: {
+    enable: {
+      doc: 'Set to enable [HSTS](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security) if HTTPS is also enabled',
+      format: 'Boolean',
+      env: 'CMD_HSTS_ENABLE',
+      default: true
+    },
+    maxAgeSeconds: {
+      doc: 'Maximum duration in seconds to tell clients to keep HSTS status',
+      format: 'nat',
+      env: 'CMD_HSTS_MAX_AGE',
+      default: 60 * 60 * 24 * 365
+    },
+    includeSubdomains: {
+      doc: 'Set to include subdomains in HSTS',
+      format: 'Boolean',
+      env: 'CMD_HSTS_INCLUDE_SUBDOMAINS',
+      default: true,
+    },
+    preload: {
+      doc: 'whether to allow preloading of the site\'s HSTS status (e.g. into browsers)',
+      format: 'Boolean',
+      env: 'CMD_HSTS_PRELOAD',
+      default: true
+    }
+  },
+  csp: {
+    enable: {
+      doc: 'whether to enable Content Security Policy.',
+      format: 'Boolean',
+      env: 'CMD_CSP_ENABLE',
+      default: true
+    },
+    directives: {
+      doc: 'Additional CSP directives to the default ones required for CodiMD to work.',
+      default: {}
+    },
+    addDefaults: {
+      doc: 'Option to control whenever the recommended CSP directives should be add.',
+      format: 'Boolean',
+      default: true
+    },
+    addDisqus: {
+      doc: 'Option to disable the addition of Disqus rules to CSP. Disabling this option will break the Disqus integration.',
+      format: 'Boolean',
+      default: true
+    },
+    addGoogleAnalytics: {
+      doc: 'Option to disable the addition of Google Analytics rules to CSP. Disabling this option will break the Google Analytics integration.',
+      format: 'Boolean',
+      default: true
+    },
+    upgradeInsecureRequests: {
+      default: 'auto'
+    },
+    reportURI: {
+      doc: 'Allows to add a URL for CSP reports in case of violations.',
+      format: '*',
+      default: undefined
+    }
```

Comment:
is it correct that all of these do not have ENV equivalents?

Reply:
For half of these, it's correct. But ReportURI has one.